apps/ai-content-generator/src/components/config/parameterReducer.ts

```diff
@@ -0,0 +1,38 @@
+import { AppInstallationParameters } from './ConfigForm';
+
+export enum ParameterAction {
+  MODEL = 'model',
+  APIKEY = 'apiKey',
+  PROFILE = 'profile',
+  CONTENTFUL_PARAMETERS = 'contentfulParameters',
+}
+
+const { MODEL, APIKEY, PROFILE, CONTENTFUL_PARAMETERS } = ParameterAction;
+
+type ParameterStringActions = {
+  type: typeof MODEL | typeof APIKEY | typeof PROFILE;
```

Comment:
So there's two ways we can do a list of enum values here:

First way is referencing the enum directly, so we have the proper typing
```js
type ParameterStringActions = {
  type: ParameterAction.MODEL | ParameterAction.APIKEY | ParameterAction.PROFILE;
...
```

Second way is if we want to destructure ParameterAction, like we are here, we have to use typeOf. Since we are using the enum typing here, not the actual value. 

Eg. if we don't use typeOf, the destructured enum's type will be unresolved/any: 
<img width="337" alt="image" src="https://github.com/contentful/apps/assets/38636581/e07793da-359a-4423-ab03-086d7d710e0d">

In the case that we use typeOf, the typeOf resolves to `ParameterAction.<EnumName>`
<img width="588" alt="image" src="https://github.com/contentful/apps/assets/38636581/6151ffac-e0b7-4fd5-8b71-18c02f06a4ea">

Reply:
I found a way to make this look cleaner than destructuring and `or`-ing the keys: we can instead use `Exclude` in TS

```ts
type ParameterStringActions = {
  type: Exclude<ParameterAction, ParameterAction.CONTENTFUL_PARAMETERS>;
  value: string;
};
type ParameterObjectActions = {
  type: ParameterAction.CONTENTFUL_PARAMETERS;
  value: AppInstallationParameters;
};
```